### DISCUSSION TITLE
why the change from `api.rootProps` to `api.getRootProps`?

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [chakra-ui/zag#1645] [Q&A] [answered]
Hi. Out of curiosity why zag-js api is changed from for example `api.rootProps` or `api.labelProps` ... to `api.getRootProps()` or `api.getLabelProps()`?

**Comment by anubra266** [comment, marked as answer]
@samadadi 
1. This improves performance, cause attributes gets lazy loaded
2. Doing this all at once, means if we ever plan to expose any props in those methods in the future, it'd no longer be a breaking change

**Reply by segunadebayo** [reply]
Spot on @anubra266 👌🏼
